File: rewrite_widget.js
const fs = require('fs');

let content = fs.readFileSync('src/components/PokethologyCombatMissionWidget.tsx', 'utf8');

// Add getDailyHubCombatChallenges import
if (!content.includes('getDailyHubCombatChallenges')) {
    content = content.replace("import { cn } from '../lib/utils';", "import { cn } from '../lib/utils';\nimport { getDailyHubCombatChallenges } from '../utils/dailyHubChallenges';");
}

fs.writeFileSync('src/components/PokethologyCombatMissionWidget.tsx', content);

File: rewrite_widget.py
import re

with open('src/components/PokethologyCombatMissionWidget.tsx', 'r') as f:
    content = f.read()

# Add getDailyHubCombatChallenges import
if 'getDailyHubCombatChallenges' not in content:
    content = content.replace("import { cn } from '../lib/utils';", "import { cn } from '../lib/utils';\nimport { getDailyHubCombatChallenges } from '../utils/dailyHubChallenges';")

with open('src/components/PokethologyCombatMissionWidget.tsx', 'w') as f:
    f.write(content)
